NPAS1.py

```diff
@@ -0,0 +1,200 @@
+        if gc_bounds[0] <= gc_percent <= gc_bounds[1] and \
+                length_bounds[0] <= seq_len <= length_bounds[1] and \
+                mean_offset >= quality_threshold:
+            filtered_seqs.append(record)
+
+    if output_filename is None:
+        output_filename = input_path.split("/")[-1]
+    else:
+        output_filename = output_filename + ".fastq"
+
+    output_path = output_filename
+    SeqIO.write(filtered_seqs, output_path, "fastq")
+
+class BiologicalSequence(ABC):
+    """
+        Abstract base class representing a biological sequence.
+
+        Defines common operations for biological sequences, such as DNA, RNA, or protein sequences.
+
+        Attributes:
+            sequence (str): The sequence of characters representing the biological sequence.
+
+        Methods:
+            __len__(): Returns the length of the biological sequence.
+            __getitem__(index): Gets the character at the specified index or returns a subsequence.
+            __str__(): Returns a string representation of the biological sequence.
+            is_valid_alphabet(): Checks if all characters in the sequence belong to a valid alphabet.
+        """
+    def __init__(self, sequence):
+        self.sequence = sequence
+        if not self.alphabet_checking():
+            raise ValueError("Invalid characters in the sequence.")
+
+    @abstractmethod
+    def __len__(self):
+        pass
+
+    @abstractmethod
+    def __getitem__(self, index):
+        pass
+
+    def __str__(self):
+        return self.sequence
+
+    def __repr__(self):
+        return f"{self.__class__.__name__}('{self.sequence}')"
+
+    def alphabet_checking(self):
+        if not set(self.sequence) <= set(type(self).ALPHABET):
+            return False
+        return True
+
+
+class NucleicAcidSequence(BiologicalSequence):
+    """
+        Represents a nucleic acid sequence, such as DNA or RNA.
+
+        Inherits from BiologicalSequence and adds functionality specific to nucleic acid sequences.
+
+        Attributes:
+            sequence (str): The sequence of characters representing the nucleic acid sequence.
+
+        Methods:
+            complement(): Returns the complement of the nucleic acid sequence.
+            gc_content(as_percentage=False): Calculates the GC content of the nucleic acid sequence.
+
+        Overrides:
+            is_valid_alphabet(): Checks if all characters in the sequence belong to a valid nucleic acid alphabet.
+        """
+
+    COMPLEMENT_DICT = {
+        'A': 'T',
+        'T': 'A',
+        'C': 'G',
+        'G': 'C',
+        'a': 't',
+        't': 'a',
+        'c': 'g',
+        'g': 'c'}
+
+    def __init__(self, sequence):
+        super().__init__(sequence)
+
+    def __len__(self):
+        return len(self.sequence)
+
+    def __getitem__(self, index):
+        return self.sequence[index]
+
+    def complement(self):
+        return ''.join(self.COMPLEMENT_DICT.get(base, base) for base in self.sequence)
```

Comment:
```suggestion
        return self.__class__(''.join(self.COMPLEMENT_DICT.get(base, base) for base in self.sequence))
```

можно как-то так попробовать, а то результат выполнения возвращает строку, что не совсем хорошо

![image](https://github.com/Python-BI-2023/Peer_review/assets/109810573/3bfb0bd4-178a-4781-874b-e5a8769554e3)

Reply:
Чаще делают не `self.__class__(...)`, а `type(self)(...)`